### ISSUE TITLE
for loop variable

### ISSUE THREAD

**Issue by JoePelz** [riolet/ritchie#16]
Question.  Should this work?
```
i = 3
i for 5, 10
  echo i + " "
```
Right now the for loop fails because i is already defined.

**Comment by JoePelz** [comment]
It should work. Needs a fix now to allow it

**Comment by JoePelz** [comment]
As of c65cbabf9f6ca03d46dd2f975821586452a3e64b, the above code compiles. 
However, this is still an issue because the following code now compiles (to uncompilable C code):
```
3 for 5, 10
  doStuff

//compiles into:
// for (3=5;3<(10);3 ++)
```

**Comment by JoePelz** [comment]
Fixed the above issue with commit 796e9876d17a27547e9cb42a353c1698aa79e3e3

This is still a band-aid on the bigger problem of differentiating between the integer `i` and the integer `3`.
